packages/components/pie-radio-group/src/index.ts

```diff
@@ -11,10 +25,133 @@ const componentSelector = 'pie-radio-group';
 
 /**
  * @tagname pie-radio-group
+ * @slot - Default slot
+ * @slot label - The label slot
+ * @event {CustomEvent} change - when one of the radios state is changed.
  */
-export class PieRadioGroup extends LitElement implements RadioGroupProps {
+export class PieRadioGroup extends FormControlMixin(RtlMixin(LitElement)) implements RadioGroupProps {
+    @state()
+        hasLabel = false;
+
+    @property({ type: String })
+    public name?: RadioGroupProps['name'];
+
+    @property({ type: String })
+    public value = defaultProps.value;
+
+    @property({ type: Boolean })
+    public isInline = defaultProps.isInline;
+
+    @property({ type: Boolean, reflect: true })
+    public disabled = defaultProps.disabled;
+
+    @queryAssignedElements({ selector: 'pie-radio' })
+        _slottedChildren: Array<HTMLInputElement> | undefined;
+
+    /**
+     * Dispatches a custom event to notify each slotted child radio element
+     * when the radio group is disabled.
+     * @private
+     */
+    private _handleDisabled (): void {
+        this._slottedChildren?.forEach((child) => child.dispatchEvent(new CustomEvent(ON_RADIO_GROUP_DISABLED, {
+            bubbles: false, composed: false, detail: { disabled: this.disabled },
+        })));
+    }
+
+    /**
+     * Unselects all radios that are not the selected value.
+     * @param {string} selectedValue - The value of the currently selected radio.
+     * @private
+     */
+    private handleRadioSelection (selectedValue: string): void {
+        this._slottedChildren?.filter((radio) => !radio.disabled)
+        .forEach((radio) => {
+            if (radio.value === selectedValue) {
+                this.value = selectedValue;
+                radio.checked = true;
+            } else {
+                radio.checked = false;
+            }
+        });
+    }
+
+    /**
+     * Handles the radio change event, updates the selected value, and emits a change event for consumers.
+     * @param {Event} event - The change event from a radio element.
+     * @private
+     */
+    private _handleRadioChange (event: Event): void {
+        event.stopPropagation();
+        const target = event.target as HTMLInputElement;
+        this.handleRadioSelection(target.value);
+        const changedEvent = wrapNativeEvent(event);
+        this.dispatchEvent(changedEvent);
+    }
+
+    /**
+     * Updates the `hasLabel` state when content is added to the label slot.
+     * @param {Event} e - The slotchange event.
+     * @private
+     */
+    private handleSlotChange (e: { target: HTMLSlotElement }): void {
+        const childNodes = e.target.assignedNodes({ flatten: true });
```

Comment:
Can you use the `queryAssignedNodes` decorator for this? It also takes `flatten` as an option and then you might be able to avoid needing to listen to the `slotchange` event entirely and just have a getter for `_hasLabel` which checks the length of the assigned nodes array